Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -529,8 +529,13 @@ public final class PackageBuilder {
 
         /// Returns the path of the given target.
         func findPath(for target: TargetDescription) throws -> AbsolutePath {
-            // If there is a custom path defined, use that.
-            if let subpath = target.path {
+            if target.type == .binary {
+                if let artifact = self.binaryArtifacts.first(where: { $0.path.basenameWithoutExt == target.name }) {
+                    return artifact.path
+                } else {
+                    throw ModuleError.artifactNotFound(target.name)
+                }
```

Comment:
nit:

```
guard let artifact = self.binaryArtifacts.first(where: { $0.path.basenameWithoutExt == target.name }) else {
  throw ModuleError.artifactNotFound(target.name)
}
return artifact.path
```

Reply:
`Xcode` has almost pulled `conversion to guard expression` off :D